fix(control): add react/ink type dependencies for typecheck

## Background
Discovered during QA of #231. The `packages/control` package is missing `react` and `ink` type declarations, causing `bun typecheck` (which runs `tsc -b`) to fail on every commit. The pre-commit hook runs typecheck, so this blocks all commits repo-wide.

## Problem
All `.tsx` and hook files in `packages/control/src/` produce errors like:
- `Cannot find module 'react' or its corresponding type declarations`
- `Cannot find module 'ink' or its corresponding type declarations`
- `Parameter 'prev' implicitly has an 'any' type`

This has been a pre-existing issue — the control package compiles at runtime via Bun but not via `tsc -b`.

## Proposed Fix
Either:
1. Add `@types/react`, `react`, and `ink` as devDependencies in the control package (or root)
2. Or exclude `packages/control` from `tsc -b` if it's not meant to be typechecked via tsc

## Relevant Files
- `packages/control/tsconfig.json`
- `packages/control/package.json`
- Root `tsconfig.json` (references)

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
   "devDependencies": {
     "@biomejs/biome": "^1.9.4",
     "@types/bun": "latest",
+    "@types/react": "^18.3.0",
     "typescript": "^5.9.3"
   },
   "dependencies": {

